### DISCUSSION TITLE
pic16f controller interfacing with rfm98

### DISCUSSION THREAD

**Discussion by Sumya99** [jgromes/RadioLib#723] [Q&A]
Hello all, 
I am just trying to implement spi write and read from rfm98 using pic16f1789 series of controller. For example in the picture, it just request is read command(0b**0** A6A5A4 A3A2A1A0) and chipVersion register which is (0x42) but is gives zero. I am only connecting SPI pins( MOSI MISO SCK SS).

I have tested following: 
1. My radio module is fine working using arduino and the RadioLib
2. Logic voltage level from PIC16f1789 is good (~3.3V)
3. Wiring test's done with multimeter and it is ok
4. I set the spi clock same as of RadioLib

What could be the potential causes?  Please help me.

Thanks in advance :)
![dslabchipversion](https://user-images.githubusercontent.com/48245515/230705360-e58f67f0-3164-45c1-b2f8-d19c0ca70d96.png)

**Comment by jgromes** [comment]
Unfortunately I don't have much experience with PIC16 MCUs. The traces look OK, maybe try recording for RadioLib as well and see if there's some differences e.g. in timing?

**Reply by jgromes** [reply]
FYI you can port RadioLib on any platform, even non-Arduino ones: https://github.com/jgromes/RadioLib/wiki/Porting-to-non-Arduino-Platforms
